Base/Crypto/Crypto.Static/RSAKeyManager.hpp

```diff
@@ -0,0 +1,218 @@
+        return publicKeyStr;
+    };
+
+    /// @brief Method for loading key pair
+    void loadKeyPair() { loadPrivateKey(); };
+
+    /** @brief Method for saving rsa keys in files
+     * @details Work steps: \
+     * 1. Save private key. If saving the private key fails, the public key is not saved too.
+     * 2. Save public key.
+     */
+    void saveKeyPair()
+    {
+        if (savePrivateKey() == GeneralCodes::SUCCESS)
+        {
+            savePublicKey();
+        }
+    };
+
+    /// @brief Getter for private server key
+    CryptoPP::RSA::PrivateKey getPrivateKey() { return _keyPair.privateKey; };
+
+private:
+    /// Contains public and private keys
+    RSAKeyPair                          _keyPair;
+    /// Name of file to save private RSA key
+    std::string                         PRIVATE_KEY_FILE{"juniorgram.rsa"};
+    /// Name of file to save public RSA key
+    std::string                         PUBLIC_KEY_FILE{"juniorgram.rsa.pub"};
+
+    /** @enum ValidationLevel
+    * @details Determines, which checks will be performed by CryptoPP library. \
+    * For more info: https://www.cryptopp.com/wiki/Keys_and_Formats#Validating_Keys
+    */
+    enum ValidationLevel : uint32_t
+    {
+        TRIFLING = 0,
+        MINOR,
+        BALANCED,
+        HARD
+    };
+
+    /** @brief Method for loading private key from a file
+    * @details Method loads private RSA key from file "PRIVATE_KEY_FILE". The following situations are possible: \
+    * - File not found and private key not loaded -> generate new key pair \
+    * - Private key loaded, but invalid -> generate new key pair \
+    * - Private key loaded and valid -> load public key
+    */
+    inline void loadPrivateKey()
+    {
+        bool privateKeyLoaded = false;
+
+        std::ifstream privateKeyOutputFile(PRIVATE_KEY_FILE);
+        if (privateKeyOutputFile.is_open())
+        {
+            privateKeyOutputFile.close();
+
+            FileSource privateKeySource(PRIVATE_KEY_FILE.c_str(), true);
+            _keyPair.privateKey.BERDecode(privateKeySource);
+            privateKeyLoaded = true;
+        }
+        privateKeyOutputFile.close();
+
+        if (privateKeyLoaded)
+        {
+            if (AutoSeededRandomPool randPool; _keyPair.privateKey.Validate(randPool, ValidationLevel::HARD))
+            {
+                Base::Logger::FileLogger::getInstance().log("[SERVER] Private key loaded from file successfully",
+                                                            Base::Logger::LogLevel::INFO);
+                loadPublicKey();
+            }
+            else
+            {
+                FileLogger::getInstance().log("[SERVER] Invalid (probably compromised) private key", Base::Logger::LogLevel::WARNING);
+                _keyPair = RSAKeyGenerator().generateRSAKeyPair();
+            }
+        }
+        else
+        {
+            FileLogger::getInstance().log("[SERVER] Private key not founded", Base::Logger::LogLevel::INFO);
```

Comment:
"[SERVER] Private key not found_ed_"
need: "[SERVER] Private key not found"
If it doesn't mean something else.